homeassistant/components/synology_dsm/binary_sensor.py

```diff
@@ -68,3 +76,12 @@ class SynoDSMStorageBinarySensor(SynologyDSMDeviceEntity, BinarySensorEntity):
     def is_on(self) -> bool:
         """Return the state."""
         return getattr(self._api.storage, self.entity_type)(self._device_id)
+
+
+class SynoDSMUpgradeBinarySensor(SynologyDSMEntity, BinarySensorEntity):
+    """Representation a Synology Upgrade binary sensor."""
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state."""
+        return getattr(self._api.upgrade, self.entity_type)
```

Comment:
```suggestion

    @property
    def available(self) -> bool:
    """Return True if entity is available."""
        return bool(self._api.upgrade)
```

Reply:
I don't understand your suggestion @Quentame

Reply:
I´m not sure, if you realy want me to remove the return statement of is_on() 😄 
Property available() is added 😉

Reply:
Ups indeed I missed the is_on line that you should keep.

It was only to add `available`